modules/sdk/node-build/esbuild/writeImportBridges.mjs

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs/promises';
+import path from 'path';
+
+import getFlatName from '../util/getFlatName.mjs';
+import getExportedSymbols from '../util/getExportedSymbols.mjs';
+import getImportBridgePath from './getImportBridgePath.mjs';
+
+export default async function writeImportBridges(globalImports, globalSymbols) {
+	return Promise.all(
+		Object.entries(globalImports)
+			.filter(([moduleName, {external}]) => external && !moduleName.endsWith('.css'))
+			.map(
+				([moduleName, {webContextPath}]) => 
+					writeImportBridge(globalSymbols, moduleName, webContextPath)
+			)
+	);
+}
+
+async function writeImportBridge(globalSymbols, moduleName, webContextPath) {
+	const importBridgePath = getImportBridgePath(moduleName);
+
+	let symbols;
+
+	try {
+		symbols = getExportedSymbols(moduleName);
+	}
+	catch(error) {
+		symbols = globalSymbols[moduleName];
+
+		if (symbols === undefined) {
+			console.error(
+				`❌ Cannot infer exported symbols for ${moduleName}:`,
+				`please define them in the liferay-portal/modules/node-build.config.js file.\n`);
```

Comment:
love this. Any way we could document _how_ we would actually do that? Like if I am a a dev getting this error, how do I determine what symbol to add to the config?

Reply:
There's no easy way. If this error shows it's because the module cannot be inspected with `require()`. That means it is not suitable for server development and the only way to know what it will really export is by reading the docs or inspecting the source code.

This problem existed in npm-scripts too.